python/runtime/feature/compile.py

```diff
@@ -0,0 +1,164 @@
+
+
+def compile_feature_column(fc, model_type, package):
+    """
+    Compile a IR FeatureColumn object to a runtime feature column object.
+
+    Args:
+        fc (FeatureColumn): the IR FeatureColumn object.
+        model_type (EstimatorType): one of TENSORFLOW and XGBOOST.
+        package (module): the Python package corresponding to the model_type.
+
+    Returns:
+        A runtime feature column object.
+    """
+    fc_package = package.feature_column
+
+    if isinstance(fc, NumericColumn):
+        fd = fc.get_field_desc()[0]
+        nc = fc_package.numeric_column(fd.name, shape=fd.shape)
+        return nc
+
+    if isinstance(fc, BucketColumn):
+        source_fc = compile_feature_column(fc.source_column, model_type,
+                                           package)
+        bc = fc_package.bucketized_column(source_fc, boundaries=fc.boundaries)
+        return bc
+
+    if isinstance(fc, CategoryIDColumn):
+        fd = fc.get_field_desc()[0]
+        if fd.vocabulary:
+            return fc_package.categorical_column_with_vocabulary_list(
+                key=fd.name, vocabulary_list=list(fd.vocabulary))
+        else:
+            return fc_package.categorical_column_with_identity(
+                key=fd.name, num_buckets=fc.bucket_size)
+
+    if isinstance(fc, SeqCategoryIDColumn):
+        assert model_type != EstimatorType.XGBOOST, \
+            "SEQ_CATEGORY_ID is not supported in XGBoost models"
+        fd = fc.get_field_desc()[0]
+        return fc_package.sequence_categorical_column_with_identity(
+            key=fd.name, num_buckets=fc.bucket_size)
+
+    if isinstance(fc, CategoryHashColumn):
+        fd = fc.get_field_desc()[0]
+        dtype = to_package_dtype(fd.dtype, package)
+        return fc_package.categorical_column_with_hash_bucket(
+            key=fd.name, hash_bucket_size=fc.bucket_size, dtype=dtype)
+
+    if isinstance(fc, CrossColumn):
+        assert model_type != EstimatorType.XGBOOST, \
+            "CROSS is not supported in XGBoost models"
+        key_strs = []
+        for key in fc.keys:
+            if isinstance(key, six.string_types):
+                key_strs.append(key)
+            elif isinstance(key, NumericColumn):
+                fd = key.get_field_desc()[0]
+                size = np.prod(fd.shape) if fd.shape else 1
+                assert size == 1, "CROSS does not support shape not equal to 1"
+                key_strs.append(fd.name)
+            else:
+                raise ValueError(
+                    "field in CROSS must be of FeatureColumn or string type")
+
+        return fc_package.crossed_column(key_strs,
+                                         hash_bucket_size=fc.hash_bucket_size)
+
+    if isinstance(fc, EmbeddingColumn):
+        assert model_type != EstimatorType.XGBOOST, \
+            "EMBEDDING is not supported in XGBoost models"
+        cc = compile_feature_column(fc.category_column, model_type, package)
```

Comment:
What does cc stand for?

Reply:
I have replaced `cc` with `category_column` for a better review.